Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js

```diff
@@ -0,0 +1,153 @@
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+const COMMIT_URL = 'https://commits.webkit.org/';
+const VALID_REF_RE = /^[a-zA-Z0-9\/\.\@-]+$/;
+const ERROR_TEMPLATE = `<div class="text small failed"> {message} </div>`;
+
+commitInputs = document.getElementsByClassName('commitInput');
+Array.from(commitInputs).forEach((input) => {
+    input.addEventListener('change', ValidateCommitFormat);
+    input.addEventListener("keyup", ({key}) => {
+        if (key === "Enter") {
+            document.getElementById(`${input.name}Button`).click();
+        }
+    })
+});
+
+async function GetCommit(commit) {
+    const response = await fetch(`/${commit}/json`);
+    if (!response.ok) {
+        throw new RangeError(`Could not find ${commit}.<br>Please fix any typos or check if this commit exists.`);
+    } else {
+        const commitObj = await response.json();
+        return commitObj;
+    }
+}
+
+function ValidateCommitFormat(){
+    const commitRep = this.value;
+
+    const messages = document.getElementsByName('message');
+    let message;
+    for (const value of messages.values()) {
+        if (`${this.name}Message` == value.id) {
+            message = value;
+            break;
+        }
+    }
+
+    if (VALID_REF_RE.test(commitRep) || !commitRep) {
+        message.innerHTML = '';
+        document.getElementById(this.id).classList.remove('invalid');
+    } else {
+        message.innerHTML = `<div class="text small failed">${commitRep} is not a valid commit format</div>`;
+        document.getElementById(this.id).classList.add('invalid');
+    }
+}
+
+async function GenerateCompareLink() {
+    const commitOne = document.getElementById('commit1').value;
+    const commitTwo = document.getElementById('commit2').value;
+    let message = document.getElementById('compareMessage');
+
+    if (!commitOne | !commitTwo) {
+        message.innerHTML = ERROR_TEMPLATE.replace("{message}", "Must provide two commits to compare");
```

Comment:
Instead of using replace, you can make you template a function, pass in message.
```javascript
function errorMessage(message) {
    return `<div class="text small failed"> ${message} </div>`;
}
```